.github/workflows/pull_requests.yml

```diff
@@ -68,4 +68,33 @@ jobs:
 
         for target in "${targets[@]}"; do
           swift package plugin generate-documentation --target "$target" --warnings-as-errors --analyze --level detailed
-        done
\ No newline at end of file
+        done
+
+  acceptable-language-check:
+    name: Acceptable language check
+    runs-on: ubuntu-latest
+    timeout-minutes: 1
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+    - name: Run acceptable language check
+      shell: bash
+      run: |
+        set -euo pipefail
+
+        log() { printf -- "** %s\n" "$*" >&2; }
+        error() { printf -- "** ERROR: %s\n" "$*" >&2; }
+        fatal() { error "$@"; exit 1; }
+
+        UNACCEPTABLE_WORDS="blacklist|whitelist|slave|master|sane|sanity|insane|insanity|kill|killed|killing|hang|hung|hanged|hanging"
+
+        log "Checking for unacceptable language..."
+        PATHS_WITH_UNACCEPTABLE_LANGUAGE=$(git grep \
+          -l -F -w -E ${UNACCEPTABLE_WORDS} \
+        ) || true | /usr/bin/paste -s -d " " -
+
+        if [ -n "${PATHS_WITH_UNACCEPTABLE_LANGUAGE}" ]; then
+          fatal "❌ Found unacceptable language in files: ${PATHS_WITH_UNACCEPTABLE_LANGUAGE}."
+        fi
```

Comment:
We can pull out the configuration into a required environment variable which will make it easier to factor this into its own action later. It also uses space separated, rather than forcing the configuration to use regex syntax.

Additionally, this uses some different `grep` flags which make sure we see the filename, line, column, and unacceptable term (not the whole line), and case-insensitive matching.

```suggestion
       test -n "${UNACCEPTABLE_LANGUAGE_PATTERNS:-}" || fatal "UNACCEPTABLE_LANGUAGE_PATTERNS unset"
       
       log "Checking for unacceptable language..."
       UNACCEPTABLE_LANGUAGE_LINES=$(git -C "${REPO_ROOT}" grep \
         -w -i -I -o \
         -H -n --column \
         -E "${UNACCEPTABLE_LANGUAGE_PATTERNS// /|}" \
       ) || true | /usr/bin/paste -s -d " " -
       
       if [ -n "${UNACCEPTABLE_LANGUAGE_LINES}" ]; then
         fatal " ❌ Found unacceptable language:
       ${UNACCEPTABLE_LANGUAGE_LINES}
       "
       fi
```

This should result in the following (tested as a standalone script in a project I have with some manual examples added). It will give one output line for each violation and in a form that could later be consumed to comment back to the PR.

```console
❯ env UNACCEPTABLE_LANGUAGE_PATTERNS="kill hanging" bash scripts/check-for-unacceptable-language.sh
** Checking for unacceptable language...
** ERROR:  ❌ Found unacceptable language:
Sources/PetstoreConsumerTestCore/Common.swift:179:1:Kill
Sources/_OpenAPIGeneratorCore/FeatureFlags.swift:35:17:hanging
Sources/_OpenAPIGeneratorCore/FeatureFlags.swift:36:11:kill
Sources/swift-openapi-generator/FilterCommand.swift:97:1:kill
```